x-pack/plugins/ml/public/application/trained_models/models_management/test_models/models/ner/ner.tsx

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import React, { FC, useState, useMemo } from 'react';
+
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { EuiSpacer, EuiTextArea, EuiButton, EuiTabs, EuiTab } from '@elastic/eui';
+
+import { useMlApiContext } from '../../../../../contexts/kibana';
+import { NerInference } from './ner_inference';
+import type { FormattedNerResp } from './ner_inference';
+import { NerOutput } from './ner_output';
+import { MLJobEditor } from '../../../../../jobs/jobs_list/components/ml_job_editor';
+import { extractErrorMessage } from '../../../../../../../common/util/errors';
+import { ErrorMessage } from '../../inference_error';
+import { OutputLoadingContent } from '../../output_loading';
+
+interface Props {
+  model: estypes.MlTrainedModelConfig;
+}
+
+enum TAB {
+  TEXT,
+  RAW,
+}
+
+export const NerModel: FC<Props> = ({ model }) => {
+  const { trainedModels } = useMlApiContext();
+
+  const ner = useMemo(() => new NerInference(trainedModels, model), [trainedModels, model]);
+
+  const [inputText, setInputText] = useState('');
+  const [isRunning, setIsRunning] = useState(false);
+  const [output, setOutput] = useState<FormattedNerResp | null>(null);
+  const [rawOutput, setRawOutput] = useState<string | null>(null);
+  const [selectedTab, setSelectedTab] = useState(TAB.TEXT);
+  const [showOutput, setShowOutput] = useState(false);
+  const [errorText, setErrorText] = useState<string | null>(null);
+
+  async function run() {
+    setShowOutput(true);
+    setOutput(null);
+    setRawOutput(null);
+    setIsRunning(true);
+    setErrorText(null);
+    try {
+      const { response, rawResponse } = await ner.infer(inputText);
+      setOutput(response);
+      setRawOutput(JSON.stringify(rawResponse, null, 2));
+    } catch (e) {
+      setIsRunning(false);
+      setOutput(null);
+      setErrorText(extractErrorMessage(e));
```

Comment:
Is it worth checking the length of text entered, as with even a few lines of text I am getting a `timeout [10s] waiting for inference result` error.

Reply:
Increasing to 30s has helped on my setup, although with long passages of text I am finding the network request stays in a pending state and never seems to time out.